src/admm.jl

```diff
@@ -0,0 +1,139 @@
+"""
+        admm(
+            m::AbstractBlockNLPModel;
+            kwargs... 
+        )
+Solves a `BlockNLPModel` with the ADMM algorithm.
+
+# Arguments
+
+- `m::AbstractBlockNLPModel`: identifier for the `BlockNLPModel` 
+- `kwargs...`: options for the solver
+"""
+function admm(
+    m::AbstractBlockNLPModel;
+    option_dict::Dict{Symbol,Any} = Dict{Symbol,Any}(),
+    kwargs...,
+)
+    start_time = time()
+    nb = m.problem_size.block_counter # Number of blocks
+    iter_count = 0
+
+    # check if warm start primal-dual solutions are available 
+    # otherwise initialize them with zero vectors
+    if :primal_start ∈ keys(kwargs)
+        x = Vector{Float64}(kwargs[:primal_start])
+        pop!(kwargs, :primal_start)
+    else
+        x = zeros(Float64, m.problem_size.var_counter)
+    end
+    if :dual_start ∈ keys(kwargs)
+        y = Vector{Float64}(kwargs[:dual_start])
+        pop!(kwargs, :dual_start)
+    else
+        y = zeros(Float64, n_constraints(m))
+    end
+
+    opt = Options(primal_start = x, dual_start = y)
+    set_options!(opt, option_dict, kwargs)
+
+    tired = false
+    converged = false
+
+    if opt.verbosity > 0
+        @info log_header(
+            [:iter, :objective, :Ax_b, :λ, :elapsed_time, :max_block_time],
```

Comment:
Regarding the log, I think we are almost there. I would suggest a few renamings, to improve clarity: 
- `:Ax_b` -> `|Ax -b| ` or `inf_pr`
- `:lambda` -> `inf_du` (if we choose to be consistent with Ipopt)

But can we specify the header as string (and not symbol)?